### PR TITLE
feat: Add support for kpt live apply

### PR DESCRIPTION
**Pull request by msvticket**
Also cleaning up a bit

Using this feature would fix jenkins-x/jx#8024

Note: Since kpt is very strict about the yaml syntax there is still one more fix needed for kpt to work:
cdfoundation/tekton-helm-chart#28

Depends on jenkins-x-plugins/jx-gitops#891

### CODE REVIEW

**Comment by ankitm123** [comment]
We also need to test it on existing jx installations. But we are not making it default yet, right? opt in for now?

**Comment by msvticket** [comment]
Yeah, the default stays the same: `KUBEAPPLY ?= kubectl-apply`

So you would opt in with `KUBEAPPLY = kpt-apply` in your `Makefile`.

**Comment by msvticket** [comment]
So I have run this in our dev cluster since I created this PR. It's working fine, but it's slower than `kubectl apply`. Especially reconciliation of CRDs are slow. 

An option would be to have separate kpt packages for cluster, customresourcedefinitions and namespaces and apply them separately (as for kubectl apply) but also adding the functionality to only apply a package if it is updated. What do you say? I'll try testing that in our dev cluster.

**Comment by msvticket** [comment]
A problem with that is that nothing will be applied when running on a new cluster if config-root is already populated.

**Comment by msvticket** [comment]
It turns out that changing the package structure is tricky since moving resources in a cluster from one package (resource group) to another isn't straight forward to automate until GoogleContainerTools/kpt#2488 is solved. So we need to be pretty sure about the which packages to have before merging...

**Comment by msvticket** [comment]
I think I have addressed the issues I mentioned above.

/hold cancel

File: charts/cdf/tekton-pipeline/defaults.yaml
Comment (ankitm123) [inline charts/cdf/tekton-pipeline/defaults.yaml:3]: Is this required for this change? I would prefer to keep it in different PRs.

Comment (msvticket) [reply charts/cdf/tekton-pipeline/defaults.yaml:3]: Yes.
"Since kpt is very strict about the yaml syntax there is still one more fix needed for kpt to work"
Without cdfoundation/tekton-helm-chart#28 I got problems with duplicate entries in a configmap due to https://github.com/jenkins-x/jx3-versions/blob/master/charts/cdf/tekton-pipeline/values.yaml.gotmpl#L25-L27

Of course I could downgrade the tekton version in cdfoundation/tekton-helm-chart to get a release with my fix but still tekton version 0.29.

Comment (ankitm123) [reply charts/cdf/tekton-pipeline/defaults.yaml:3]: Ah yes, I remember. ok we should first get 0.1.0 out and then do this IMO.

**Comment by msvticket** [comment]
> A possible work around could be to not use `--service-side`. I'm testing that now.

Well, it might work when starting from scratch, but not when already having the problem.

**Comment by msvticket** [comment]
@ankitm123 I think it's time to get this PR merged...

**Comment by babadofar** [comment]
I have no objections to the contents of this PR

**Review by ankitm123 — APPROVED** [review]

**Comment by msvticket** [comment]
I guess I need to retest until reaching an unused build number for this PR. Which should be 8.
/retest

**Comment by msvticket** [comment]
the next should get build number 8:
/retest

**Pull request merged**
